apps/balancer-tools/src/app/apr/(utils)/calculatePoolStats.ts

```diff
@@ -38,71 +37,68 @@ const WEEKS_IN_YEAR = 52;
 const SECONDS_IN_DAY = 86400;
 const SECONDS_IN_YEAR = 365 * SECONDS_IN_DAY;
 
-const fetchPoolAveragesForDateRange = withCache(
-  async function fetchPoolAveragesInRangeFn(
-    poolId: string,
-    network: string,
-    from: number,
-    to: number,
-  ): Promise<[number, number, string, { symbol: string; balance: string }[]]> {
-    const res = await pools.gql(network).poolSnapshotInRange({
+async function fetchPoolAveragesForDateRange(
+  poolId: string,
+  network: string,
+  from: number,
+  to: number,
+): Promise<[number, number, string, { symbol: string; balance: string }[]]> {
+  const res = await pools.gql(network).poolSnapshotInRange({
+    poolId,
+    from,
+    to,
+  });
+
+  res.poolSnapshots = res.poolSnapshots.sort(
+    (a, b) => a.timestamp - b.timestamp,
+  );
+
+  if (res.poolSnapshots.length === 0) {
+    // Following Fabio's recomendation on #BAL-872
```

Comment:
This came from a rebase, I could have removed/added something that I shouldn't have here @Rawallon @luizakp

Reply:
tested here, everything working fine